qualtran/bloqs/chemistry/trotter.py

```diff
@@ -0,0 +1,553 @@
+#  Copyright 2023 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from functools import cached_property
+from typing import Dict, Optional, Set, Tuple
+
+import numpy as np
+from attrs import frozen
+from cirq_ft import TComplexity
+from cirq_ft.algos.qrom import QROM
+from cirq_ft.infra.bit_tools import float_as_fixed_width_int
+from numpy.typing import NDArray
+
+from qualtran import Bloq, BloqBuilder, Register, Signature, SoquetT
+from qualtran.bloqs.arithmetic import (
+    Add,
+    MultiplyTwoReals,
+    OutOfPlaceAdder,
+    ScaleIntByReal,
+    SquareRealNumber,
+    SumOfSquares,
+)
+from qualtran.bloqs.basic_gates import Rz, TGate
+from qualtran.bloqs.basic_gates.rotation import RotationBloq
+from qualtran.cirq_interop import CirqGateAsBloq
+
+
+def get_polynomial_coeffs() -> Tuple[NDArray, NDArray]:
+    """Polynomial coefficients from Fusion paper.
+
+    Useful for tests and notebook so add as function.
+    """
```

Comment:
spruce this up


 - longer, more explicit function name to hint at its example nature
 - "Fusion paper" -> something better
 - "so add as function" -> fix

Reply:
Done.